paasta_tools/setup_kubernetes_job.py

```diff
@@ -149,22 +150,26 @@ def setup_kube_deployments(
     api_updates = 0
     for _, app in applications:
         if app:
-            if (
-                app.kube_deployment.service,
-                app.kube_deployment.instance,
-            ) not in existing_apps:
-                log.info(f"Creating {app} because it does not exist yet.")
-                app.create(kube_client)
-                api_updates += 1
-            elif app.kube_deployment not in existing_kube_deployments:
-                log.info(f"Updating {app} because configs have changed.")
-                app.update(kube_client)
-                api_updates += 1
-            else:
-                log.info(f"{app} is up-to-date!")
-
-            log.info(f"Ensuring related API objects for {app} are in sync")
-            app.update_related_api_objects(kube_client)
+            try:
+                if (
+                    app.kube_deployment.service,
+                    app.kube_deployment.instance,
+                ) not in existing_apps:
+                    log.info(f"Creating {app} because it does not exist yet.")
+                    app.create(kube_client)
+                    api_updates += 1
+                elif app.kube_deployment not in existing_kube_deployments:
+                    log.info(f"Updating {app} because configs have changed.")
+                    app.update(kube_client)
+                    api_updates += 1
+                else:
+                    log.info(f"{app} is up-to-date!")
+
+                log.info(f"Ensuring related API objects for {app} are in sync")
+                app.update_related_api_objects(kube_client)
+            except Exception:
+                log.error(f"Error while processing: {app}")
+                log.error(traceback.format_exc())
```

Comment:
```suggestion
                log.exception(f"Error while processing: {app}")
```

will clean this up a bit and mean that we don't have to import/use `traceback`

Reply:
Thanks for the suggestions